core/src/Kokkos_HIP_Space.hpp

```diff
@@ -99,6 +99,15 @@ class HIPSpace {
                   const size_t arg_logical_size = 0) const;
 
  private:
+  /** Note: if we're moving HIP out of Experimental,
+   *  and you're getting an error regarding LogicalMemorySpace
+   *  not being a class, look at how we solved this in CudaSpace.
+   *  Basically, you need to forward declare LogicalMemorySpace
+   *  in namespace Experimental above this class, and namespace
+   *  qualify LogicalMemorySpace here
+   */
```

Comment:
Comment outdated by moving the forward declaration to `<Kokkos_Core_fwd.hpp>`

Reply:
Handled via 85bb2f3